tests/playbooks/python_distribution.yaml

```diff
@@ -131,7 +159,89 @@
+        name: test_python_distribution_pull_through
+        base_path: pull_through_mirror
+        remote: test_python_remote_pull_through
+        state: present
+      register: result
+    - name: Verify distribution of pull-through mirror
+      ansible.builtin.assert:
+        that:
+          - result.changed == true
+          - result.distribution.name == "test_python_distribution_pull_through"
+          - result.distribution.base_path == "pull_through_mirror"
+          - ansible_check_mode or result.distribution.base_url is regex(".*/pypi/pull_through_mirror")
+          - result.distribution.remote == remote_result.remote.pulp_href
+
+    - name: Create distribution for pull-through mirror (2nd try)
+      pulp.squeezer.python_distribution:
+        name: test_python_distribution_pull_through
+        base_path: pull_through_mirror
+        state: present
+      register: result
+    - name: Verify distribution of pull-through mirror (2nd try)
+      ansible.builtin.assert:
+        that:
+          - result.changed == false
+
+    - name: Read distribution
+      pulp.squeezer.python_distribution:
+        name: test_python_distribution_pull_through
+      register: result
+    - name: Verify read distribution
+      ansible.builtin.assert:
+        that:
+          - result.changed == false
+          - result.distribution.name == "test_python_distribution_pull_through"
+          - result.distribution.base_path == "pull_through_mirror"
+          - result.distribution.base_url is regex(".*/pypi/pull_through_mirror")
+          - result.distribution.remote == remote_result.remote.pulp_href
+
+    - name: Remove remote from pull-through mirror distribution
+      pulp.squeezer.python_distribution:
+        name: test_python_distribution_pull_through
+        base_path: pull_through_mirror
+        remote: ""
+        state: present
+      register: result
+    - name: Verify remove remote from pull-through mirror distribution
+      ansible.builtin.assert:
+        that:
+          - result.changed == true
+          - result.distribution.name == "test_python_distribution_pull_through"
+          - result.distribution.base_path == "pull_through_mirror"
+          - ansible_check_mode or result.distribution.base_url is regex(".*/pypi/pull_through_mirror")
+          - result.distribution.remote == None
+
+    - name: Delete distribution
+      pulp.squeezer.python_distribution:
+        name: test_python_distribution_pull_through
+        state: absent
+      register: result
+    - name: Verify delete distribution
+      ansible.builtin.assert:
+        that:
+          - result.changed == true
+          - not result.distribution
+
+    - name: Delete distribution (2nd try)
+      pulp.squeezer.python_distribution:
+        name: test_python_distribution_pull_through
+        state: absent
+      register: result
+    - name: Verify delete distribution (2nd try)
+      ansible.builtin.assert:
```

Comment:
I don't think we need those. Deletion of a distribution is already covered before.

Reply:
This deletes a a distribution with a remote, the one before does not have a remote. But I can remove this test here.

Reply:
Yes, think of it that way: These tests should demonstrate that squeezer can talk to the api. It's not about testing the Pulp server. And the api call for deletion is simply the same. So if this can drop 2 seconds of every test run, i would prefer to have it. Also can you squash the commits in the end?

Reply:
Fixed the tests. And I squashed some of the commits. Do you want them to be squashed altogether? Cause they do different things.